packages/snap/src/polkadot/api.ts

```diff
@@ -22,9 +23,11 @@ async function initApi(wsRpcUrl: string): Promise<ApiPromise> {
 }
 
 export const getApi = async (wallet: Wallet): Promise<ApiPromise> => {
-  if (!api) {
+  const config = getConfiguration(wallet);
+  // api not initialized or configuration changed
+  if (!api || rpcUrl != config.wsRpcUrl) {
```

Comment:
This is what I was referring to today that API wasn't reinitializing on network change.

Reply:
doesn't api already have some way to fetch rpc url. If not, it's better to just set api to null when configure method is called

Reply:
Generally open PR with just fix like that, because we know have to wait until this PR got merged to have that fixed. If possible, try to cherry pick commit with this fix or create another branch and pr with this fix.